hack/dep-update.sh

```diff
@@ -29,7 +29,7 @@ done
     echo $_sync_only
     cd staging/src/kubevirt.io/client-go
     if [ "${_sync_only}" == "false" ]; then go get $@ ./...; fi
-    go mod tidy
+    go mod tidy -compat=1.17
```

Comment:
Can you explain this change?

Reply:
@xpivarc Thanks! @machadovilaca I think I commented on it in the previous review cycle, why did you still keep it?

Reply:
my bad, sorry, this was indeed resolved in a previous interaction, but after some reviews eventually it was changed back because we then discussed some unnecessary dependency updates in `staging/src/kubevirt.io/client-go/go.mod`, namely, `onsi/ginkgo/v2` and `onsi/gomega`, that didn't seem necessary

by default the flag defaults to ensure compatibility with the previous version (`-compat=1.16` since this `go.mod` declares go 1.17) which causes dependency incompatibilities and required manual updates of many dependencies, this change guaranties compatibility with declared go 1.17

an alternative to `-compat=1.17` addition would be adding on top off this PR changes this commit: https://github.com/kubevirt/kubevirt/commit/f348aca1bac1b63809f1965482983fa2b254fb8a

@enp0s3 @xpivarc

Reply:
The required changes look good.

Reply:
added